File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
n1, n2 = map(int, input().split())
a, b = max(n1, n2), min(n1, n2)
while True:
    if a % b == 0:
        break
    else:
        a, b = b, a % b
print(b)
print(n1 * n2 // b)

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int a = Math.min(n, m), b = Math.max(n, m);
        while (b % a != 0) {
            int temp = a;
            a = b % a;
            b = temp;
        }
        b = n * m / a;
        return new int[] {a, b};
    }
}